packages/cli/src/cmds/agentInstaller/projectConfiguration.ts

```diff
@@ -81,12 +86,47 @@ async function resolveSelectedInstallers(
     }
   }
 
-  // Sanity check. Verify everything resolved properly. This should branch should never occur.
+  // Sanity check. Verify everything resolved properly. This branch should never occur.
   if (projects.some(({ selectedInstaller }) => !selectedInstaller)) {
     throw new Error('Invalid selection');
   }
 }
 
+export async function getYarnPackages(availableInstallers: AgentInstaller[]): Promise<string[]> {
+  let yarnPackages: string[] = [];
+  for (const installer of availableInstallers) {
+    if (installer.name === 'yarn') {
+      const packageJsonFilename = join(installer.path, 'package.json');
+      if (await exists(packageJsonFilename)) {
+        const data = await fs.readFile(packageJsonFilename, 'utf-8');
+        const packageJson = JSON.parse(data);
+        if ('workspaces' in packageJson) {
+          const workspaces = packageJson['workspaces'];
+          // process as glob because it could be something like
+          // workspaces: [ 'packages/*' ]
+          for (const workspace of workspaces) {
+            const matches = glob.sync(workspace, {
+              matchBase: true,
+              cwd: installer.path,
+              ignore: ['**/node_modules/**', '**/.git/**'],
+              strict: false,
+            });
+            for (const yarnPackage of matches) {
+              // packages must be directories; don't break on leftover files
+              const isDir = lstatSync(join(installer.path, yarnPackage)).isDirectory();
+              if (isDir) {
+                yarnPackages.push(yarnPackage);
+              }
+            }
+          }
```

Comment:
I don't think this is a good method of doing this. Not all directories matching the globs will be workspaces. You should instead call `yarn workspaces list --json` and parse the output — as a bonus, you'll get the root workspace in the list and will be able to handle it the same way as the others.

Reply:
With yarn 1.22.19 the workspaces command fails...

```
$ yarn workspaces list --json
yarn workspaces v1.22.19
error Invalid subcommand. Try "info, run"
info Visit https://yarnpkg.com/en/docs/cli/workspaces for documentation about this command.

$ yarn run workspaces list --json
yarn run v1.22.19
error Command "workspaces" not found.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

...even though it's present in the list of available commands
```
$ yarn help | grep -i workspace
    --focus                             Focus on a single workspace by installing remote copies of its sibling workspaces.
    - workspace
    - workspaces
```

Installed newer version of yarn with
```
npm install -g yarn@2.4.3
```

With yarn 2.4.3
```
$ yarn workspaces list --json
{"location":".","name":"yarn_with_subprojects"}
{"location":"packages/subproject_one","name":"subproject_one"}
{"location":"packages/subproject_two","name":"subproject_two"}
```

--

Maybe we need to differentiate between using yarn 1 and yarn >=  2?
- yarn 1 can use the existing logic
- yarn 2 can use `yarn workspaces list --json`

Reply:
I implemented detecting yarn 1 and yarn >=2.  Now changing the testcases.

Reply:
The tests pass locally but not on CI.

Reply:
The tests now pass on CI.